src/includes_normalize-win32.cc

```diff
@@ -28,10 +28,13 @@ namespace {
 
 bool InternalGetFullPathName(const StringPiece& file_name, char* buffer,
                              size_t buffer_length, string *err) {
-  DWORD result_size = GetFullPathNameA(file_name.AsString().c_str(),
-                                       buffer_length, buffer, NULL);
+  wchar_t w_buffer[_MAX_PATH];
+  wstring w_file_name = Utf8ToWide(file_name.AsString());
+  DWORD result_size =
+      GetFullPathName(w_file_name.c_str(), buffer_length, w_buffer, NULL);
+  strcpy(buffer, WideToUtf8(w_buffer).c_str());
```

Comment:
what if the result of WideToUtf8 is longer than buffer_length?

Also using std::copy instead of strcpy would be good here (no need for c_str() and null termination).

Reply:
Bit unsure on the std::copy and how it should be integrated here. My suggestion is to change the sequence to 
` string s_buffer = WideToUtf8(w_buffer);`
`  if (s_buffer.size() > buffer_length) {`
 `   *err = "path too long";`
`    return false;`
 ` }`
to ensure that the converted string fits into the buffer.

Reply:
std::string::size() doesn't include the null termination byte, so you'll have to +1/-1 in your if.

One way of copying the string (haven't tested):

```
buffer[s_buffer.copy(buffer, s_buffer.size())] = '\0';
```